pkg/tests/api/alerting/api_provisioning_test.go

```diff
@@ -809,3 +813,70 @@ func createTestRequest(method string, url string, user string, body string) *htt
 	}
 	return req
 }
+
+func TestIntegrationExportFileProvision(t *testing.T) {
+	testinfra.SQLiteIntegrationTest(t)
+
+	dir, p := testinfra.CreateGrafDir(t, testinfra.GrafanaOpts{
+		DisableLegacyAlerting: true,
+		EnableUnifiedAlerting: true,
+		DisableAnonymous:      true,
+		AppModeProduction:     true,
+	})
+
+	provisioningDir := filepath.Join(dir, "conf", "provisioning")
+	alertingDir := filepath.Join(provisioningDir, "alerting")
+	err := os.MkdirAll(alertingDir, 0750)
+	require.NoError(t, err)
+
+	grafanaListedAddr, env := testinfra.StartGrafanaEnv(t, dir, p)
+
+	apiClient := newAlertingApiClient(grafanaListedAddr, "admin", "admin")
+	createUser(t, env.SQLStore, env.Cfg, user.CreateUserCommand{
+		DefaultOrgRole: string(org.RoleAdmin),
+		Password:       "admin",
+		Login:          "admin",
+		IsAdmin:        true,
+	})
+
+	apiClient.ReloadCachedPermissions(t)
+	t.Run("when provisioning alert rules from files", func(t *testing.T) {
+		// add file provisioned alert rules
+		fileProvisionedAlertRules, err := testData.ReadFile(path.Join("test-data", "provisioning-rules.yaml"))
+		require.NoError(t, err)
+
+		var expected definitions.AlertingFileExport
+		require.NoError(t, yaml.Unmarshal(fileProvisionedAlertRules, &expected))
+		expectedYaml, err := yaml.Marshal(expected)
+		require.NoError(t, err)
+
+		// create folder
+		folderUID := "my_first_folder_uid"
+		apiClient.CreateFolder(t, folderUID, "my_first_folder_with_$escaped_symbols")
+
+		err = os.WriteFile(filepath.Join(alertingDir, "provisioning-rules.yaml"), fileProvisionedAlertRules, 0750)
```

Comment:
I wonder if we should clean up after this test, so it won't affect other tests

Reply:
I figured it didn't matter really for these rn, but I can add that to my follow up PR